### ISSUE TITLE
narrowing to `Never` from instance of metaclass

### ISSUE THREAD

**Issue by beauxq** [microsoft/pyright#6152]
**Describe the bug**
pyright incorrectly narrows type to `Never` when narrowing from instance of metaclass

**Code or Screenshots**

```python
import abc
from typing import ClassVar
from typing_extensions import reveal_type


class M(abc.ABCMeta):
    pass


class B(abc.ABC, metaclass=M):
    pass


class C(B):
    x: ClassVar[tuple[int, int]] = (0, 1)


def f(m: M) -> None:
    if issubclass(m, B):
        reveal_type(m)
        if issubclass(m, C):
            reveal_type(m)
            for i in m.x:
                print(i)


f(C)
```

output:
```
Runtime type is 'M'
Runtime type is 'M'
0
1
```
as expected

mypy:
```
Revealed type is "Type[test.B]"
Revealed type is "Type[test.C]"
```
as expected

pyright:
```
20:21 - information: Type of "m" is "Never"
22:25 - information: Type of "m" is "Never"
23:22 - error: "Never" is not iterable
```

**VS Code extension or command-line**
>Are you running pyright as a VS Code extension, a language server in another editor, integrated into Pylance, or the command-line tool?

both

> Which version?

pylance v2023.10.20
pyright v1.1.277

I don't know what version this is using:
https://pyright-playground.decorator-factory.su/?gzip=H4sIAMqLKWUC_22OTQrCMBSE9-8Ub5lA9QAFBZN1vEJJ64sEmh-SKHp7Y0tLxc5y5vtgrIshFdT9ACYFh-Udrb939Crksw0-o52JRE_SY1d3AoBh1DmjYtU7XoRUVDRvAWtiHVZALECDriJTeVJ7pGTit76RQcNci4rj4YzX4GnerUGb86OfPOYaXMRvNi-Z42v958iNs-MBGCY5fAD4DxbiHQEAAA%3D%3D

**Comment by erictraut** [comment]
Thanks for the bug report. This will be addressed in the next release.

**Comment by beauxq** [comment]
I don't very often use the command line version, most of my pyright usage is through pylance.

**Comment by erictraut** [comment]
This is included in pyright 1.1.332, which I just published. It will also be included in a future release of pylance.
